Pcp 231 en tant qu utilisateur je peux faire une demande d oublie de mot de passe afin de recevoir un email pour changer de mot de passe

Maxwellmilien opened the pull request.

Maxwellmilien committed:
feat: update dockerfile for email-service

Maxwellmilien committed:
feat: add Email Service Dependencies

Maxwellmilien committed:
feat: set minimal for run email server

Maxwellmilien committed:
feat: add email-service in lint-staged

Maxwellmilien committed:
feat: add email-service in service Makefile

Maxwellmilien committed:
feat: dockerfile add cmd run dev

Maxwellmilien committed:
feat: add email-service in docker-compose dev

Maxwellmilien committed:
feat: add email service in nginx.conf

Maxwellmilien committed:
feat: add /reset route

Maxwellmilien committed:
feat: email-service add nodeemailer

Maxwellmilien committed:
feat: change nginx email-service

Maxwellmilien committed:
feat: add resetPassword controller

Maxwellmilien committed:
feat: add dotenv/config

Maxwellmilien committed:
feat: add mailer service

Maxwellmilien committed:
feat: rename controller to controllers

Maxwellmilien committed:
feat: add getTemplate in utils

Maxwellmilien committed:
feat: add jsonwebtoken, handlebars

Maxwellmilien committed:
feat: add view reset

Maxwellmilien committed:
feat: add type payload

Maxwellmilien committed:
feat: add SMTP varaible in docker-compose

Maxwellmilien committed:
feat: remove async from getTemplate

Maxwellmilien committed:
feat: send email in reset-password controller

Maxwellmilien committed:
feat: Add JWT validation to reset password endpoint

Add a new secret key for email service JWT validation
Add middleware to validate JWT before password reset

Maxwellmilien committed:
feat: Add axios for email service integration & Add password reset functionality

Maxwellmilien committed:
feat: Add forgot password functionality to login form

Maxwellmilien committed:
feat: Add winston logging and error handling

Maxwellmilien committed:
feat: remove message return in error middleware

Maxwellmilien committed:
feat: add email in payload reset

Maxwellmilien committed:
feat:updateAdd Joi validation to reset password controller
Add middleware to check payload origin
Update error handling to send status codes
Update package dependencies
Rename reset-password controller file

Maxwellmilien committed:
feat: add toast in loginpage

Yiroma commented on backend/auth-service/src/controllers/resetController.ts:
```diff
@@ -0,0 +1,28 @@
+
+async function resetController(req: Request, res: Response) {
+  const { error } = resetPasswordSchema.validate(req.body);
+  if (!error) {
+    try {
+      const userId = await getIdByEmailRepository(req.body.email);
+      const token = createJwtRestForEmailService(userId, req.body.email);
+      await emailClient.get('/reset', {
+        headers: {
+          Authorization: `Bearer ${token}`,
+          'Content-Type': 'application/json',
+        },
+      });
+    } catch (error) {
+      res.sendStatus(204); //202 ?
+      return;
+    }
+  }
+
+  res.sendStatus(204); //202 ?
```
Ne te manque t'il pas des return ?
```js
if (!error) {
  try {
    ...
    return res.sendStatus(202); // Demande accepté (envoi d'email en cours...)
  } catch (error) {
    return res.sendStatus(500); // Erreur interne du serveur
  }
}

return res.sendStatus(400); // données invalides
```

Maxwellmilien replied:
non, non la fonction ne retourne rien !

Yiroma commented on lint-staged.config.js:
```diff
@@ -13,6 +13,13 @@ module.exports = {
       `prettier --write ${paths.join(' ')}`,
     ];
   },
+  'email/*/src/**/*.{js,jsx,ts,tsx}': (filenames) => {
+    const paths = filenames.map((filename) => `"${filename}"`);
+    return [
+      `eslint --fix --rule 'no-console: [error, { allow: ["warn", "error", "info"] }]' ${paths.join(' ')}`,
+      `prettier --write ${paths.join(' ')}`,
+    ];
+  },
```
Je ne pense pas que ce soit utile car juste au dessus on demande de contrôler tout les dossiers dans backend et de contrôler à l'intérieur de chaqu'un des dossiers src tout les fichier js,ts.

Yiroma commented on frontend/src/components/LoginRoot/LoginRoot.tsx:
```diff
@@ -129,6 +142,17 @@ export default function LoginRoot() {
             </div>
           )}
         </form>
+        <div className="mt-4 w-full text-right">
```
div inutile. tu peux directement applique ce style dans ton button

Yiroma commented on frontend/src/components/LoginRoot/LoginRoot.tsx:
```diff
@@ -129,6 +142,17 @@ export default function LoginRoot() {
+        <div className="mt-4 w-full text-right">
+          <button
+            type="button"
+            className="forgot-password-link bg-transparent border-none p-0 cursor-pointer"
```
la valeur `forgot-password-link` n'existe pas. à retiré

className corrigé avec ajout de valeur de la div:
```js
className="w-full mt-4 p-2 text-sm text-right text-turquoise-500 hover:text-turquoise-600 bg-transparent border-none cursor-pointer"
```

Maxwellmilien committed:
feat: delete email in mginx

Maxwellmilien committed:
feat: set resetUrl in playload to email service

Maxwellmilien committed:
fix: delete div

Maxwellmilien committed:
fix: delete email lint-stage

Maxwellmilien committed:
fix: styling of forgot password button

Maxwellmilien committed:
feat: style

Maxwellmilien committed:
feat: add test integration

Maxwellmilien committed:
feat: add error test integration

Maxwellmilien committed:
feat: add test integration bad origin service

Maxwellmilien committed:
fix: delete code

Maxwellmilien committed:
fix: change secret key for email

Maxwellmilien committed:
fix: dockerfile

Maxwellmilien committed:
fix: email docker copy config jest

Maxwellmilien committed:
feat: email reset password page

Maxwellmilien committed:
fix: change get reset by pass

Maxwellmilien committed:
fix: change return 400 by 422 to check payload

Maxwellmilien committed:
feat: Update JWT middleware error handling and payload cleanup

Update error code for missing secret server key
Use bracket notation for deleting payload properties

Maxwellmilien committed:
fix: change import order

Maxwellmilien committed:
fix: coment + fix test

Maxwellmilien commented on backend/email-service/src/controllers/reset.controller.ts:
```diff
@@ -0,0 +1,45 @@
+
+async function resetPassword(req: Request, res: Response) {
+  if (!process.env.SECRET_KEY_TOKEN_EMAIL) {
+    throw new HttpError(500, 'Missing secret email key');
```
vérifier code

Maxwellmilien commented on backend/email-service/src/controllers/reset.controller.ts:
```diff
@@ -0,0 +1,45 @@
+      html: template({ url: `${payloadAuth.resetUrl}?token=${token}` }),
+    };
+
+    await transporter.sendMail(mailOptions);
```
add log avec logger

Maxwellmilien committed:
feat: update

Maxwellmilien committed:
fix: change name html to hbs

Maxwellmilien committed:
fix: error code

Maxwellmilien committed:
fix: reset form

Maxwellmilien committed:
fix: use logger in index.ts

Maxwellmilien committed:
feat: add router

Maxwellmilien committed:
fix: logging send email change

Maxwellmilien committed:
fix: route reset to get

Maxwellmilien committed:
fix: move component

Maxwellmilien committed:
fix: import

Maxwellmilien committed:
fix: body change to content

Yiroma commented on frontend/src/components/user/RestPassword.tsx:
```diff
@@ -0,0 +1,52 @@
+        })}
+      />
+      <Button type="submit">Envoyer email</Button>
+    </form>
```
Ajouter éventuellement un bouton de retour.
Modifier le terme du bouton : "Réinitialiser mon mot de passe"

Maxwellmilien replied:
Pour le moment j'ai écrit uniquement Réinitialiser dans le bouton sinon ca casse l’affichage :disappointed:

Yiroma commented on frontend/src/components/user/RestPassword.tsx:
```diff
@@ -0,0 +1,52 @@
+    >
+      <InputLabel
+        type="email"
+        label="Adresse email de recupération"
```
Accent : "Adresse email de récupération"

Yiroma commented on frontend/src/components/user/RestPassword.tsx:
```diff
@@ -0,0 +1,52 @@
+  return (
+    <form
+      className="w-full max-w-sm flex flex-col items-center gap-4"
+      aria-label="Formulaire de d'envoie d'un email de réinitialisation de mot de passe"
```
Terme à modifier : "Formulaire de réinitialisation de mot de passe"

Yiroma commented on backend/email-service/src/views/reset.view.hbs:
```diff
@@ -0,0 +1,2 @@
+<h2>Rest password</h2>
```
Titre à mettre en Français

eustasio reviewed (dismissed):
c'est bon pour. moi code approuvé et test feature ok

Maxwellmilien committed:
feat: add button Annuler in RestPassword

Maxwellmilien committed:
fix: translate

Maxwellmilien committed:
fix: button name

Maxwellmilien committed:
fix: annuler to retour

eustasio reviewed (dismissed):
Si on a le temps fix du bug visuel sur le boutton retour lors du clique

Maxwellmilien committed:
fix: size button retour

Maxwellmilien committed:
Merge branch 'dev' into PCP-231-En-tant-qu-utilisateur-je-peux-faire-une-demande-d-oublie-de-mot-de-passe-afin-de-recevoir-un-email-pour-changer-de-mot-de-passe

Maxwellmilien committed:
fix: fix merge

Maxwellmilien committed:
fix: fix merge

Maxwellmilien committed:
fix add hbs to .prettierignore
